src/dune_rules/coq/coq_rules.ml

```diff
@@ -163,23 +163,94 @@ let theories_flags ~theories_deps =
      Command.Args.S (List.map ~f:theory_coqc_flag libs))
 ;;
 
-let boot_flags ~coq_lang_version t : _ Command.Args.t =
-  let boot_flag = if coq_lang_version >= (0, 8) then [ "-boot" ] else [] in
-  match t with
-  (* We are compiling the prelude itself
-     [should be replaced with (per_file ...) flags] *)
-  | `Bootstrap_prelude -> As ("-noinit" :: boot_flag)
-  (* No special case *)
-  | `No_boot | `Bootstrap _ -> As boot_flag
-;;
+module Bootstrap : sig
+  type t =
+    | Implicit (** Use implicit stdlib loading, for coq lang versions < 0.8 *)
+    | No_stdlib
+    (** We are in >= 0.8, however the user set stdlib = no
+        , or we are compiling the prelude *)
+    | Stdlib of Coq_lib.t (** Regular case in >= 0.8 (or in < 0.8
+                              (boot) was used *)
 
-let coqc_file_flags
-  ~dir
-  ~theories_deps
-  ~wrapper_name
-  ~boot_type
-  ~ml_flags
-  ~coq_lang_version
+  val empty : t Memo.t
```

Comment:
What's the point of memoizing this here?

Reply:
No point, just to make it more uniform with the other; if we don't like this I suggest we instead refactor the point where this is needed to remove the call.